### PR TITLE
feat: Add broken up version of PCES classes

### PR DESCRIPTION
**Pull request by alittley**
- part of #10024 

The new files in this PR are a broken-up rework of the existing PCES files. The new files aren't used yet, and the old files are left alone. 

[diagram](https://mermaid.ink/svg/Zmxvd2NoYXJ0IExSCnYwKCgiKiIpKQpzdWJncmFwaCB2MVsiQ29uc2Vuc3VzIFBpcGVsaW5lIl0Kc3ViZ3JhcGggdjJbIkxpbmtlZCBFdmVudCBJbnRha2UiXQp2M3t7ImdldE1pbmltdW1HZW5lcmF0aW9uTm9uQW5jaWVudCJ9fQp2NFsibGlua2VkRXZlbnRJbnRha2UiXQp2NXt7ImxpbmtlZEV2ZW50SW50YWtlX3NwbGl0dGVyIn19CmVuZAp2NlsiaW5PcmRlckxpbmtlcjxiciAvPioiXQplbmQKc3ViZ3JhcGggdjdbIkludGFrZSBQaXBlbGluZSJdCnN1YmdyYXBoIHY4WyJFdmVudCBWYWxpZGF0aW9uIl0KdjlbImV2ZW50RGVkdXBsaWNhdG9yPGJyIC8+KiJdCnYxMFsiZXZlbnRTaWduYXR1cmVWYWxpZGF0b3I8YnIgLz4qIl0KdjExWyJpbnRlcm5hbEV2ZW50VmFsaWRhdG9yPGJyIC8+4oCgIl0KZW5kCnN1YmdyYXBoIHYxMlsiT3JwaGFuIEJ1ZmZlciJdCnYxM1sib3JwaGFuQnVmZmVyPGJyIC8+KiJdCnYxNHt7Im9ycGhhbkJ1ZmZlcl9zcGxpdHRlciJ9fQplbmQKdjE1W1siZXZlbnRIYXNoZXIiXV0KZW5kCnN1YmdyYXBoIHYxNlsiUENFUyJdCnYxN1svImV2ZW50RHVyYWJpbGl0eU5leHVzIi9dCnYxOFsvInBjZXNTZXF1ZW5jZXIiL10KdjE5WyJwY2VzV3JpdGVyPGJyIC8+KiJdCmVuZApzdWJncmFwaCB2MjBbIlN0YXRlIEZpbGUgTWFuYWdlbWVudCJdCnYyMXt7ImV4dHJhY3Qgb2xkZXN0TWluaW11bUdlbmVyYXRpb25PbkRpc2sifX0KdjIyWyJzaWduZWRTdGF0ZUZpbGVNYW5hZ2VyIl0KdjIze3sidG8gU3RhdGVXcml0dGVuVG9EaXNrQWN0aW9uIn19CmVuZAp2MjRbImV2ZW50Q3JlYXRpb25NYW5hZ2VyPGJyIC8+KuKZoSJdCnYyNVsiaGVhcnRiZWF0Il0KdjI2Wy8icGNlc1JlcGxheWVyIi9dCnYyN1sic3RhdGVTaWduZXIiXQp2MjgoKCLigKAiKSkKdjI5KCgi4pmhIikpCnYyNCAtLiAibm9uLXZhbGlkYXRlZCBldmVudHMiIC4tPiB2MjgKdjkgLS0gImV2ZW50cyB3aXRoIHVudmFsaWRhdGVkIHNpZ25hdHVyZXMiIC0tPiB2MTAKdjE1IC0tICJub24tdmFsaWRhdGVkIGV2ZW50cyIgLS0+IHYxMQp2MTAgLS0gInVub3JkZXJlZCBldmVudHMiIC0tPiB2MTMKdjIxIC0tICJtaW5pbXVtIGdlbmVyYXRpb24gdG8gc3RvcmUiIC0tPiB2MTkKdjMgLS4gIm1pbmltdW0gZ2VuZXJhdGlvbiBub24gYW5jaWVudCIgLi0+IHYwCnYyNSAtLSAiaGVhcnRiZWF0IiAtLT4gdjI5CnY2IC0tICJsaW5rZWQgZXZlbnRzIiAtLT4gdjQKdjExIC0tICJub24tZGVkdXBsaWNhdGVkIGV2ZW50cyIgLS0+IHY5CnY0IC0tPiB2NQp2NSAtLT4gdjMKdjEzIC0tPiB2MTQKdjE0IC0tICJwb3NzaWJsZSBwYXJlbnRzIiAtLT4gdjI0CnYxNCAtLSAiZXZlbnRzIHRvIHNlcXVlbmNlIiAtLT4gdjE4CnYyNiAtLSAiZXZlbnRzIHRvIGhhc2giIC0tPiB2MTUKdjI2IC0tICJkb25lIHN0cmVhbWluZyBwY2VzIiAtLT4gdjE5CnYxOCAtLSAidW5saW5rZWQgZXZlbnRzIiAtLT4gdjYKdjE4IC0tICJldmVudHMgdG8gd3JpdGUiIC0tPiB2MTkKdjE5IC0tICJsYXRlc3QgZHVyYWJsZSBzZXF1ZW5jZSBudW1iZXIiIC0tPiB2MTcKdjIyIC0tPiB2MjEKdjIyIC0tPiB2MjMKY2xhc3NEZWYgczAgZmlsbDojZjg4LHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjAsdjI4LHYyOSBzMApjbGFzc0RlZiBzMSBmaWxsOiM5Q0Ysc3Ryb2tlOiMwMDAsc3Ryb2tlLXdpZHRoOjJweApjbGFzcyB2MSx2MTYsdjIwLHY3IHMxCmNsYXNzRGVmIHMyIGZpbGw6I0JFRixzdHJva2U6IzAwMCxzdHJva2Utd2lkdGg6MnB4CmNsYXNzIHYxMix2Mix2OCBzMgpjbGFzc0RlZiBzMyBmaWxsOiNjY2Msc3Ryb2tlOiMwMDAsc3Ryb2tlLXdpZHRoOjJweApjbGFzcyB2MTQsdjE3LHYxOCx2MjEsdjIzLHYyNix2Myx2NSBzMwpjbGFzc0RlZiBzNCBmaWxsOiNmZjksc3Ryb2tlOiMwMDAsc3Ryb2tlLXdpZHRoOjJweApjbGFzcyB2MTAsdjExLHYxMyx2MTUsdjE5LHYyMix2MjQsdjI1LHYyNyx2NCx2Nix2OSBzNAo=?bgColor=e8e8e8)

### CODE REVIEW

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/EventDurabilityNexus.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+     *
+     * @param event the event in question
+     * @return true if the event is guaranteed to be durable, false otherwise
+     */
+    public boolean isEventDurable(@NonNull final GossipEvent event) {
+        Objects.requireNonNull(event);
+        if (event.getStreamSequenceNumber() == GossipEvent.STALE_EVENT_STREAM_SEQUENCE_NUMBER) {
+            // Stale events are not written to disk.
+            return false;
+        }
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/EventDurabilityNexus.java:53]: Not necessarily something we need to do in this PR: setting a different sequence number for stale events is overly complex and not really necessary. Eventually I'd like to stop doing it.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PcesFiles.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Tracks preconsensus event files currently on disk.
+ */
+public class PcesFiles {
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PcesFiles.java:36]: Nitpick on naming. In general, I find naming a class using a plural noun sometimes makes things hard to scan when you are doing your first read through. What if we called this instead something like `PcesFileTracker`?

**Review by cody-littley — COMMENTED** [review]
Overall, LGTM other than a few nitpicks. Lol, git blame is going to make people start coming to you when the PCES breaks. 😄

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PcesFiles.java (hunk 2 above)
Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PcesFiles.java]: Done

**Review by cody-littley — APPROVED** [review]

**Pull request merged**